spec/lib/message_bus_spec.rb

```diff
@@ -188,6 +188,22 @@
       site_id.must_equal 'law-and-order'
       channel.must_equal '/chuck'
     end
+
+    it "passes custom pub sub options" do
+      publish_args = nil
+
+      @bus.reliable_pub_sub.define_singleton_method :publish do |*args|
```

Comment:
@SamSaffron Are you happy with this style of stubbing?

Reply:
not really, I much prefer a full integration test here @pawelniewie hitting age and size should be fairly straight forward with the memory provider, the one thing I am perfectly comfortable mocking is time but we should have some examples doing that already imo.